doc/contribute/committer_expectations.rst

```diff
@@ -0,0 +1,165 @@
+  the change.
+
+- Easier to rebase and merge. Smaller PRs are less likely to conflict with other
+  changes in the tree.
+
+- Easier to rollback if the PR breaks functionality.
+
+
+Defining Small PRs
+==================
+
+- Small PRs should encompass one self-contained logical change.
+
+- When adding a new feature or API, the PR should address only one part of the
+  feature. In this case create an :ref:`RFC proposal <rfcs>` [#rfc]_ to describe
+  the additional parts of the feature for reviewers.
+
+- PRs should include tests for any added or changed lines of code.
+
+- PRs must update any documentation affected by the functional code changes.
+
+- If introducing a new API, the PR must include an example usage of the API.
+  This provides context to the reviewer and prevents submitting PRs with unused
+  APIs.
+
+
+Multiple Commits on a Single PR
+===============================
+
+You are further encouraged to break up PR into multiple commits.  Keep in mind
+each commit of your PR should still build cleanly and pass all the CI tests.
+
+For example, when introducing an extension to an API, you can break up the PR
+into multiple commits targeting these specific changes:
+
+#. Introduce the new APIs, including shared devicetree bindings
+#. Update driver implementation X, with driver specific devicetree bindings
+#. Update driver implementation Y
+#. Add tests for the new API
+#. Add a sample using the API
+#. Update the documentation
+
+Large Changes
+=============
+
+Large changes to the Zephyr project must submit an :ref:`RFC proposal <rfcs>`
+describing the full scope of change and future work.  The :ref:`RFC proposal
+<rfcs>` provides the required context to reviewers, but allows for smaller,
+incremental, PRs to get reviewed and merged into the project. The RFC should
+also define the minimum viable implementation.
+
+Changes which require an :ref:`RFC proposal <rfcs>` include:
+
+- Submitting a new feature.
+- Submitting a new API.
+- :ref:`treewide-changes`.
+- Other changes which encompass more than 200 lines of non-trivial modifications.
+
+PR Requirements
+***************
+
+- Each commit in the PR must provide a commit message following the
+  :ref:`commit_message_guidelines`.
+
+- All files in the PR must comply with :ref:`Licensing
+  Requirements<licensing_requirements>`.
+
+- Follow the Zephyr :ref:`coding_style`.
+
+- PRs must pass all CI checks. This is a requirement to merge the PR, but you
+  may mark a PR as draft and explicitly request reviewers to provide early
+  feedback.
+
+- When breaking a PR into multiple commits, each commit must build cleanly. This
+  is not enforced by the CI system, so it is the PR author’s responsibility to
+  verify.
+
+- When major new functionality is added, tests for the new functionality MUST be
+  added to the automated test suite. All new APIs MUST be documented and tested
+  and tests MUST cover at least 80% of the added functionality using the code
```

Comment:
Do we have some way to measure this? We used to have codecov.io, but that ended up a dud